### ISSUE TITLE
Easy display options...

### ISSUE THREAD

**Issue by NeoForce0** [microsoft/wslg#114]
**Is your feature request related to a problem? Please describe.**
One problem I ran into is running an app that opens a gui for display such as gedit. When I attempt, it gives me the following error:

(gedit:2030): Gtk-WARNING **: 13:49:50.254: cannot open display:

Then I goto stack exchange to find simple solutions to this and those didn't work for me either. I'm sure there's many other apps that can benefit from this feature.

**Comment by spronovo** [comment]
Can you take a peek at https://github.com/microsoft/wslg/wiki/Diagnosing-%22cannot-open-display%22-type-issues-with-WSLg

**Comment by pcause** [comment]
I was able to load gedit but I had problems with other apps, particularly qt apps.  I found I needed the dbus daemons (system and settings).  I posted on reddit how you can launch from your .bashrc/.bash_profile

**Comment by spronovo** [comment]
@pcause yeah, WSL distro today are typically not pre-configured for desktop usage and are either missing packages and don't launch things like dbus. Makes sense given WSL used to be a console only experience. We would like to evolve this overtime to have distro either preconfigured for WSLg or very easy to configure after install. If you have tips for getting your scenario working, please feel free to post them here and on the discussion channel in addition to reddit to help other user trying out the same scenarios.

**Comment by spronovo** [comment]
@NeoForce0 was https://github.com/microsoft/wslg/wiki/Diagnosing-%22cannot-open-display%22-type-issues-with-WSLg helpful in diagnosing your issue? Did you get things to work?

**Comment by pcause** [comment]
To get most apps running I found you have to launch dbus-daemon , both system and session. Since there is no systemd you need to create a script that gets run when wsl launches to start the dbus-daemon system daemon. Sort of like the pre-systemd days.  In my wsl.conf I run an /etc/rc.local script at launch.  This starts sshd and other thungs I like to have running.  I use this to start the system dbus:

if [ ! -e /run/dbus ];then
        mkdir /run/dbus
        chmod 755 /run/dbus
fi
/usr/bin/dbus-daemon --system --fork

To get the session dbus running in my home directory .bashrc I can check for a dbus-daemon session daemon running and if not launch , using:

if test -z "$DBUS_SESSION_BUS_ADDRESS" ; then

## if not found, launch a new one

eval `dbus-launch --sh-syntax`

echo "D-Bus per-session daemon address is: $DBUS_SESSION_BUS_ADDRESS"

fi

Might be better to start in .bash_profile.  Not sure

**Comment by pcause** [comment]
> 
> 
> @NeoForce0 was https://github.com/microsoft/wslg/wiki/Diagnosing-%22cannot-open-display%22-type-issues-with-WSLg helpful in diagnosing your issue? Did you get things to work?

I didn't see this before I got things running.  I did look at a few of the same things but didn't look at this.
